main.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from database import engine, Base, get_db
+
+Base.metadata.create_all(bind=engine)
+
+app = FastAPI()
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(author: schemas.AuthorCreate, db: Session = Depends(get_db)):
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/authors/", response_model=List[schemas.Author])
+def read_authors(skip: int = 0, limit: int = 100, db: Session = Depends(get_db)):
+    authors = crud.get_authors(db=db, skip=skip, limit=limit)
+    return authors
+
+
+@app.get("/authors/{author_id}", response_model=schemas.Author)
+def read_author(author_id: int, db: Session = Depends(get_db)):
+    author = crud.get_author(db=db, author_id=author_id)
+    if author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+    return author
+
+
+@app.post("/authors/{author_id}/books/", response_model=schemas.Book)
+def create_book_for_author(
+    author_id: int, book: schemas.BookCreate, db: Session = Depends(get_db)
+):
+    return crud.create_book(db=db, book=book, author_id=author_id)

```

Comment:
The `create_book_for_author` function should ensure that the `author_id` is correctly passed to the `create_book` function in `crud.py`. Currently, the `create_book` function in `crud.py` does not take `author_id` as a separate argument, so you may need to adjust the function signature or how the `author_id` is handled within the `create_book` function.